app/javascript/source/javascript/binx_admin.js

```diff
@@ -12,5 +14,29 @@ window.BinxAdmin = class BinxAdmin {
         .parents(".receptacle")
         .css("max-width", "100%");
     }
+    if (window.location.href.match('\\admin/graphs')){
+      this.initGraph()
+    };
+  }
+
+  initGraph() {
+    $("select#graph_date_option_choice").on("change", e => {
```

Comment:
All the things in this function other than `preventDefault` should be separated out to a separate function (which you can call with `window.binxAdmin.otherFunctionName()`) both because it will make this cleaner, but also because you should call it when the page loads (so if you're on a custom page it shows the right things)

Reply:
Okay, I think I've fixed this. I'm wondering about the .on("change") though. Part of me feels like this should go, but also I'm not sure where I'm getting the e in preventDefault() without it. You said that prevenDefault() shouldn't be pulled out of this function, so do I leave it where it is?